File: 485-max-consecutive-ones/max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int j = 0, n = nums.size();
        int current_count = 0, maxCount = INT_MIN;

        while(j < n){
            if(nums[j] == 1){
                current_count++;
            }
            else{
                maxCount = max(maxCount , current_count);
                current_count = 0;
            }
            j++;
        }
        maxCount = max(maxCount, current_count);
        return maxCount;
    }
};

File: 485-max-consecutive-ones/max-consecutive-ones.py
class Solution(object):
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n = len(nums)

        right = 0
        left = 0
        maxLength = 0

        while right < n:
            if nums[right] == 0:
                if right != n-1:
                    left = right + 1
            
            if nums[right] == 1:
                maxLength = max(maxLength, right - left + 1)
            right += 1

        
        return maxLength

            

        